build-logic/src/main/kotlin/copiedcode/CopiedCodeCheckerExtension.kt

```diff
@@ -0,0 +1,104 @@
+  /**
+   * Per-project set of additional directories to scan.
+   *
+   * This property is _not_ propagated to subprojects.
+   */
+  val scanDirectories =
+    project.objects.domainObjectContainer(
+      SourceDirectorySet::class.java,
+      { name -> project.objects.sourceDirectorySet(name, name) },
+    )
+
+  /**
+   * By default, this plugin scans all files. The content types that match the regular expression of
+   * this property are excluded, unless a content-type matches one of the regular expressions in
+   * [CopiedCodeCheckerExtension.includedContentTypePatterns].
+   *
+   * See [CopiedCodeCheckerExtension.addDefaultContentTypes],
+   * [CopiedCodeCheckerExtension.includedContentTypePatterns],
+   * [CopiedCodeCheckerExtension.includeUnrecognizedContentType],
+   * [CopiedCodeCheckerExtension.includedContentTypePatterns].
+   */
+  abstract val excludedContentTypePatterns: SetProperty<String>
+  /**
+   * By default, this plugin scans all files. The content types that match the regular expression of
+   * the [copiedcode.CopiedCodeCheckerExtension.excludedContentTypePatterns] property are excluded,
+   * unless a content-type matches one of the regular expressions in this property.
+   *
+   * See [CopiedCodeCheckerExtension.addDefaultContentTypes],
+   * [CopiedCodeCheckerExtension.excludedContentTypePatterns],
+   * [CopiedCodeCheckerExtension.includeUnrecognizedContentType],
+   * [CopiedCodeCheckerExtension.includedContentTypePatterns].
+   */
+  abstract val includedContentTypePatterns: SetProperty<String>
+
+  /**
+   * If a content-type could not be detected, this property, which defaults to `true`, is consulted.
+   *
+   * See [CopiedCodeCheckerPlugin] for details.
+   */
+  abstract val includeUnrecognizedContentType: Property<Boolean>
+
+  /**
+   * The magic "word", if present in a file, meaning "this file has been copied".
+   *
+   * A file is considered as "copied" must contain this magic word. "Word" means that the value must
+   * be surrounded by regular expression word boundaries (`\b`).
+   */
+  abstract val magicWord: Property<String>
+
+  /**
+   * License file to check, configured on the root project. See [CopiedCodeCheckerPlugin] for
+   * details.
+   */
+  abstract val licenseFile: RegularFileProperty
+
+  /** Recommended to use, adds known and used binary content types. */
+  fun addDefaultContentTypes(): CopiedCodeCheckerExtension {
+    // Exclude all images
+    excludedContentTypePatterns.add("image/.*")
+    // But include images built in XML (e.g. image/svg+xml)
+    includedContentTypePatterns.add("\\+xml")
+
+    return this
+  }
+
+  init {
+    includeUnrecognizedContentType.convention(true)
+    magicWord.convention(DEFAULT_MAGIC_WORD)
+  }
+
+  companion object {
+    // String manipulation is intentional - otherwise this source file would be considered as
+    // "copied".
+    val DEFAULT_MAGIC_WORD = "_CODE_COPIED_TO_NESSIE".substring(1)
```

Comment:
suggestion: `ALV2_REDISTRIBUTED_IN_NESSIE`

Reply:
doesn't need to be (and actually is not always) AL2

Reply:
The build checks it against the ALv2 license :shrug:

Reply:
nope - it checks whether files that have this "magic word" are mentioned in the LICENSE file

Reply:
If it were not AVv2 the mention rules would be different, right?

Reply:
I think it makes sense to link mentions to specific license type

Reply:
Maybe this comment is misplaced :) but in `[ResolvConf.java](https://github.com/projectnessie/nessie/pull/10297/files#diff-1fead81e9e321037b0c987aa7cfcbef1783dfc3ec2b4dfe50f629f55b753ab2e) for example, I think a mark of `ALV2_REDISTRIBUTED_IN_NESSIE` would be more specific to the actual license involved

Reply:
One magic word for everything